store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -299,7 +299,20 @@ public static void parseModifyVolumeRequest(ModifyVolumeRequest req) throws Serv
         // store type == 2, allow modification of only volume name
         else if (vol.getStoreType().equals(Volume.StoreType.EXTERNAL)) {
             if (volInfo.getName() != null) {
+                String storageType = "";
+                String globalBucketConfigId = "";
+                ExternalVolumeInfoHandler extVolInfoHandler = new ExternalVolumeInfoHandler(Provisioning.getInstance());
+                try {
+                    JSONObject properties = extVolInfoHandler.readServerProperties(req.getId());
+                    storageType = properties.getString(AdminConstants.A_VOLUME_STORAGE_TYPE);
+                    globalBucketConfigId = properties.getString(AdminConstants.A_VOLUME_GLB_BUCKET_CONFIG_ID);
+                } catch (JSONException e) {
+                    String errMsg = "Error while reading json data for external volume: " + req.getId();
+                    throw ServiceException.FAILURE(errMsg, e);
+                }
                 builder.setName(volInfo.getName());
+                String extRootPath = "/" + storageType + "-" + volInfo.getName() + "-" + globalBucketConfigId;
```

Comment:
what would happen if storageType Or globalBucketConfigId are empty/null?

Reply:
"storageType" should not be null/empty, but globalBucketConfigId can be(in the case of OPEN_IO, will be adding this as separate ticket). But empty/null should be added here.

Reply:
Done

Reply:
Actually there was no need of adding null check because globalBucketConfigId was initialized to empty string when object gets created